docs/source/guides/data.rst

```diff
@@ -0,0 +1,548 @@
+
+        def __getitem__(self, i):
+            return tuple(d[i] for d in self.DataSets)
+
+        def __len__(self):
+            return min(len(d) for d in self.DataSets)
+
+    class LitModel(LightningModule):
+
+        def train_dataloader(self):
+            concat_DataSet = ConcatDataSet(
+                DataSets.ImageFolder(traindir_A),
+                DataSets.ImageFolder(traindir_B)
+            )
+
+            loader = torch.utils.data.DataLoader(
+                concat_DataSet,
+                batch_size=args.batch_size,
+                shuffle=True,
+                num_workers=args.workers,
+                pin_memory=True
+            )
+            return loader
+
+        def val_dataloader(self):
+            # SAME
+            ...
+
+        def test_dataloader(self):
+            # SAME
+            ...
+
+Return multiple DataLoaders
+---------------------------
+You can set multiple DataLoaders in your :class:`~pytorch_lightning.core.lightning.LightningModule`, and Lightning will take care of batch combination.
+
+For more details please have a look at :paramref:`~pytorch_lightning.trainer.trainer.Trainer.Trainer.multiple_trainloader_mode`
+
+.. testcode::
+
+    class LitModel(LightningModule):
+
+        def train_dataloader(self):
+
+            loader_a = torch.utils.data.DataLoader(range(6), batch_size=4)
+            loader_b = torch.utils.data.DataLoader(range(15), batch_size=5)
+
+            # pass loaders as a dict. This will create batches like this:
+            # {'a': batch from loader_a, 'b': batch from loader_b}
+            loaders = {'a': loader_a,
+                       'b': loader_b}
+
+            # OR:
+            # pass loaders as sequence. This will create batches like this:
+            # [batch from loader_a, batch from loader_b]
+            loaders = [loader_a, loader_b]
+
+            return loaders
+
+Furthermore, Lightning also supports nested lists and dicts (or a combination).
+
+.. testcode::
+
+    class LitModel(LightningModule):
+
+        def train_dataloader(self):
+
+            loader_a = torch.utils.data.DataLoader(range(8), batch_size=4)
+            loader_b = torch.utils.data.DataLoader(range(16), batch_size=2)
+
+            return {'a': loader_a, 'b': loader_b}
+
+        def training_step(self, batch, batch_idx):
+            # access a dictionnary with a batch from each DataLoader
+            batch_a = batch["a"]
+            batch_b = batch["b"]
+
+
+.. testcode::
+
+    class LitModel(LightningModule):
+
+        def train_dataloader(self):
+
+            loader_a = torch.utils.data.DataLoader(range(8), batch_size=4)
+            loader_b = torch.utils.data.DataLoader(range(16), batch_size=4)
+            loader_c = torch.utils.data.DataLoader(range(32), batch_size=4)
+            loader_c = torch.utils.data.DataLoader(range(64), batch_size=4)
+
+            # pass loaders as a nested dict. This will create batches like this:
+            loaders = {
```

Comment:
Can we at least at one level convert the dict to a list to show that these can also be mixed?
If yes, the data calls in the trainstep would have to be adjusted as well :)

Reply:
feel free to suggest a fix i'm not sure exactly what you mean!

Reply:
@edenlightning done (see comment few lines below) :)